sql/statesql/statesql.go

```diff
@@ -0,0 +1,54 @@
+package statesql
+
+import (
+	"embed"
+
+	"github.com/spacemeshos/go-spacemesh/sql"
+)
+
+//go:embed schema/schema.sql schema/migrations/*.sql
+var embedded embed.FS
+
+// Database represents a state database.
+type Database struct {
+	*sql.Database
+}
+
+// Schema returns the schema for the state database.
+func Schema() (*sql.Schema, error) {
+	migrations, err := sql.LoadSQLMigrations(embedded)
+	if err != nil {
+		return nil, err
+	}
+	// NOTE: coded state migrations can be added here
+	// They can be a part of this statesql package
+	return sql.LoadSchema(embedded, migrations)
+}
```

Comment:
For instance this becomes:

```suggestion
// Schema returns the schema for the state database.
func Schema() (*sql.Schema, error) {
	sqlMigrations, err := sql.LoadSQLMigrations(migrations)
	if err != nil {
		return nil, err
	}
	// NOTE: coded state migrations can be added here
	// They can be a part of this statesql package
	return &sql.Schema{Script: schema, Migrations: sqlMigrations}, nil
}
```

Reply:
Done